router/batchrouter/asyncdestinationmanager/eloqua/manager.go

```diff
@@ -0,0 +1,50 @@
+package eloqua
+
+import (
+	"encoding/base64"
+	stdjson "encoding/json"
+	"fmt"
+
+	"github.com/rudderlabs/rudder-go-kit/bytesize"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+)
+
+func NewManager(destination *backendconfig.DestinationT) (*EloquaBulkUploader, error) {
+	destConfig := DestinationConfig{}
+	jsonConfig, err := stdjson.Marshal(destination.Config)
+	if err != nil {
+		return nil, fmt.Errorf("error in marshalling destination config: %v", err)
+	}
+	err = stdjson.Unmarshal(jsonConfig, &destConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error in unmarshalling destination config: %v", err)
+	}
+	authorization := destConfig.CompanyName + "\\" + destConfig.UserName + ":" + destConfig.Password
+	destName := destination.DestinationDefinition.Name
+	encodedAuthorizationString := "Basic " + base64.StdEncoding.EncodeToString([]byte(authorization))
+	eloquaData := HttpRequestData{
+		Authorization: encodedAuthorizationString,
+	}
+	eloqua := NewEloquaImpl("2.0")
+	baseEndpoint, err := eloqua.GetBaseEndpoint(&eloquaData)
+	if err != nil {
+		return nil, fmt.Errorf("error in getting base endpoint: %v", err)
```

Comment:
need metrics here?

Reply:
@ItsSudip I believe we have addressed this. Am I right ?

Reply:
[Minor] Can we address this ? @ItsSudip